src/main/jbake/content/blog/2016/07/14/hawkular-apm-openshift.adoc

```diff
@@ -0,0 +1,123 @@
+= Monitoring Application Performance within Openshift
+Gary Brown
+2016-07-14
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, btm, apm, distributed tracing, application performance, business transactions, openshift, cloud, docker
+
+This post will show how to use the Hawkular APM project, previously known as Hawkular BTM, to monitor an application running within Openshift. To achieve this both the Hawkular APM server, and the application (a vert.x Javascript example), will be deployed as Docker images.
+
+We will use the Openshift CLI, and therefore will need to login. For example, to login into the Openshift CDK environment:
+
+----
+oc login 10.1.2.2:8443 -u openshift-dev -p devel
+----
+
+=== Deploy Hawkular APM Server
+
+The first step is to deploy the Hawkular APM server. There are two ways that this can be achieved.
+
+==== Image from Docker Hub
+
+The Hawkular APM server has a docker image registered at https://hub.docker.com/r/jboss/hawkular-apm-server[Docker Hub]. This can be deployed to Openshift using the following commands:
+
+----
+oc new-app jboss/hawkular-apm-server
+oc expose service hawkular-apm-server
+----
+
+The _new-app_ command deploys the image as a service, and the _expose_ command makes it available via a stable URI.
+
+NOTE: If the deploy fails with an "unknown manifest" error in the openshift events log, this is due to a http://lists.openshift.redhat.com/openshift-archives/users/2016-June/msg00054.html[known docker issue related to a change between version 1.9 and 1.10]. In this case, use the local deploy approach described below.
+
+==== Local Image
+
+To deploy the APM server from a local build, it will be necessary to get the https://github.com/jboss-dockerfiles/hawkular/blob/8ffb752b68c0f49a2150fe345e660c53c32fb954/hawkular-apm-server/Dockerfile[_Dockerfile_]. Place this file in a suitable folder and from that location run the following:
+
+----
+oc new-build --binary --name=hawkular-apm-server
+oc start-build hawkular-apm-server --from-dir=. --follow
+oc new-app hawkular-apm-server
+oc expose service hawkular-apm-server
+----
+
+==== In the Openshift Console
```

Comment:
It's minor but formatting wise I would use === so that it doesn't look like a 3rd deployment option. (I slightly got confused)